File: generated_test_scripts/testcase/test_case_2.py

from selenium import webdriver
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import time
import sys

class LoginPage:
    def __init__(self, driver):
        self.driver = driver
    
    def login(self, username, password):
        WebDriverWait(self.driver, 10).until(EC.presence_of_element_located((By.XPATH, '//*[@id="user-name"]'))).send_keys(username)
        WebDriverWait(self.driver, 10).until(EC.presence_of_element_located((By.XPATH, '//*[@id="password"]'))).send_keys(password)
        WebDriverWait(self.driver, 10).until(EC.presence_of_element_located((By.XPATH, '//*[@id="login-button"]'))).click()

def run_test():
    options = Options()
    options.add_argument('--headless')
    options.add_argument('--disable-gpu')
    options.add_argument('--incognito')
    options.add_argument('--disable-notifications')

    driver = webdriver.Chrome(options=options)

    try:
        driver.get('https://saucedemo.com/')
        time.sleep(5)
        driver.maximize_window()
        
        login_page = LoginPage(driver)
        login_page.login('standard_user', 'secret_sauce')

        time.sleep(3)

        WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.XPATH, '//*[@id="add-to-cart-sauce-labs-bike-light"]'))).click()
        time.sleep(3)
        WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.XPATH, '//*[@id="add-to-cart-sauce-labs-bolt-t-shirt"]'))).click()
        time.sleep(3)

        cart_count = WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.XPATH, '//*[@id="shopping_cart_container"]/a/span')))

        if cart_count.text == '2':
            sys.exit(0)
        else:
            sys.exit(1)
    except Exception as e:
        print(f"Test failed due to {e}")
        sys.exit(1)
    finally:
        driver.quit()

run_test()

File: generated_test_scripts/poc_data - sheet1 (1)/test_case_2.java

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;
import org.apache.log4j.Logger;

public class LoginPageTest {
    private WebDriver driver;
    private static final Logger logger = Logger.getLogger(LoginPageTest.class);

    @BeforeClass
    public void setUp() {
        System.setProperty("webdriver.chrome.driver", "path/to/chromedriver");
        ChromeOptions options = new ChromeOptions();
        options.addArguments("--headless");
        options.addArguments("--disable-notifications");
        options.addArguments("--incognito");
        driver = new ChromeDriver(options);
        driver.manage().window().maximize();
        logger.info("Browser maximized and running in headless incognito mode");
    }

    @Test
    public void testInvalidUsernameLogin() {
        try {
            driver.get("https://yourwebsite.com/login");
            Thread.sleep(5000);
            logger.info("Page opened");

            WebElement usernameField = waitForElement(By.xpath("//input[@name='username']"));
            usernameField.sendKeys("incorrectUser");
            logger.info("Entered invalid username");

            WebElement passwordField = waitForElement(By.xpath("//input[@name='password']"));
            passwordField.sendKeys("Password123");
            logger.info("Entered password");

            WebElement submitButton = waitForElement(By.xpath("//button[@type='submit']"));
            submitButton.click();
            logger.info("Submit button clicked");

            WebElement errorMessage = waitForElement(By.xpath("//div[contains(text(),'Invalid username')]"));
            Assert.assertTrue(errorMessage.isDisplayed(), "Error message is not displayed for invalid username");
            logger.info("Error message verified");

            System.exit(0);
        } catch (Exception e) {
            logger.error("Test failed", e);
            System.exit(1);
        }
    }

    private WebElement waitForElement(By locator) throws InterruptedException {
        for (int i = 0; i < 5; i++) {
            try {
                WebElement element = driver.findElement(locator);
                if (element.isDisplayed()) {
                    Thread.sleep(3000); // Wait for stability
                    return element;
                }
            } catch (Exception e) {
                Thread.sleep(1000);
            }
        }
        throw new RuntimeException("Element not found: " + locator.toString());
    }

    @AfterClass
    public void tearDown() {
        if (driver != null) {
            driver.quit();
        }
    }
}
